resources/js/features/root/components/header/FacilityControl.tsx

```diff
@@ -49,45 +35,37 @@ export const FacilityControl: Component = () => {
   );
 };
 
+// TODO: it just works, may be wrong. Maybe there is a better way of handling 'global mutable state' :D
 const useFacilityControl = () => {
   const params = useParams();
   const queryClient = useQueryClient();
 
   const facilitiesQuery = createQuery(() => System.facilitiesQueryOptions());
   const statusQuery = createQuery(() => User.statusQueryOptions());
 
-  const facilities = createMemo(() =>
-    facilitiesQuery.data?.filter((facility) =>
-      statusQuery.data?.members.find(
-        (member) => member.facilityId === facility.id
-      )
-    )
+  const facilities = createMemo(
+    () =>
+      facilitiesQuery.data?.filter(
+        (facility) => statusQuery.data?.members.find((member) => member.facilityId === facility.id),
+      ),
   );
 
   createEffect(() => {
     if (params.facilityUrl) {
-      const facilities = queryClient.getQueryData<FacilityResource[]>(
-        System.facilitiesQueryOptions().queryKey
-      );
-      const facility = facilities?.find(
-        (facility) => facility.url === params.facilityUrl
-      );
+      const facilities = queryClient.getQueryData<FacilityResource[]>(System.facilitiesQueryOptions().queryKey);
```

Comment:
Tak jak mówiłem w innym miejscu (którego nie mogę teraz znaleźć, ech, paskudne jest to narzędzie do review), takie coś jest dla mnie OK tylko jeśli to wyciągniesz do jakieś funkcji utilsowej, i opiszesz dlaczego to zawsze zadziała.